syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -0,0 +1,73 @@
+import torch
+import syft
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+
+from typing import Union
+
+
+class FalconHelper:
+    XOR_EXPECTED_RING_SIZE = 2
+
+    @classmethod
+    def unfold(cls, image, kernel_size, padding):
+        return cls.__switch_public_private(
+            image, cls.__public_unfold, cls.__private_unfold, kernel_size, padding
+        )
+
+    @staticmethod
+    def __public_unfold(image, kernel_size, padding):
+        image = image.double()
+        image = torch.nn.functional.unfold(image, kernel_size=kernel_size, padding=padding)
+        image = image.long()
+        return image
+
+    @staticmethod
+    def __private_unfold(image, kernel_size, padding):
+        return image.unfold(kernel_size, padding)
+
+    @staticmethod
+    def xor(
+        value: Union[int, ReplicatedSharingTensor], other: Union[int, ReplicatedSharingTensor]
+    ) -> ReplicatedSharingTensor:
+        """
+        Compute the XOR value between value and other
+
+        value, other: values to apply the xor operation on
+
+        values, other can have the following types:
+        (ReplicatedSharingTensor ring size 2, ReplicatedSharingTensor ring size 2)
+        (Public bit {0, 1}, RST) or (RST, Public bit {0, 1})
+
+        Returns:
+            The XOR computation between value and other
+        """
+
+        if not any(isinstance(x, ReplicatedSharingTensor) for x in (value, other)):
```

Comment:
this expression will work on ints, this expression adds not-needed overhead

Reply:
We should not use ```FalconHelper``` for simple ints. We should use ```int_a``` ^ ```int_b```

Reply:
It is a check to make sure we are doing everything correctly - we could remove this after we finish the implementation.